torch/_inductor/codegen/cpp.py

```diff
@@ -2757,11 +2757,30 @@ class CppTile2DKernel(CppVecKernel):
 
     overrides = CppTile2DOverrides  # type: ignore[assignment]
 
-    def __init__(self, args, num_threads, tiling_factor, tiling_indices, tiling_dtype):
+    def __init__(
+        self,
+        args,
+        num_threads,
+        tiling_factor,
+        tiling_indices,
+        tiling_dtype,
+        inner_tail_size=None,
+        outer_tail_size=None,
+    ):
         super().__init__(
-            args, num_threads, tiling_factor, tiling_indices[1], tiling_dtype
+            args,
+            num_threads,
+            tiling_factor,
+            tiling_indices[1],
+            tiling_dtype,
+            inner_tail_size,
         )
         self.tiling_indices = tiling_indices
+        self.inner_tail_size = inner_tail_size
+        self.outer_tail_size = outer_tail_size
```

Comment:
Can we add assertion checks here to make sure `inner_tail_size` and `outer_tail_size` not dynamic?

Reply:
I checked `has_free_symbols `in `CppVecKernelChecker`. for dynamic shapes, `could_masked_vec ` is False and the tail loop still uses scalar kernel.